### ISSUE TITLE
Error: preg_replace() expects at least 3 arguments, 2 given

### ISSUE THREAD

**Issue by MarcelWeidum** [craftcms/generator#1]
### Description
When making a test plugin with the command `php craft make plugin` I get the following error:
Error: preg_replace() expects at least 3 arguments, 2 given

### Steps to reproduce

1. Install generator `composer require craftcms/generator --dev`
2. Make plugin `php craft make plugin`

### Additional info

- Craft version: 4.3.5
- PHP version: 8.0.7
- Database driver & version: MySQL 5.5.5
- Plugins & versions: Commerce 4.2.4

![Screenshot 2022-12-14 om 15 15 18](https://user-images.githubusercontent.com/9413586/207619507-12d9ed48-4a4c-4126-b3bd-4b3815b09c83.png)

**Comment by MarcelWeidum** [comment]
If has something to do with the Craft Cms version, when you fill in 4.0.0 if has the error.

**Comment by khalwat** [comment]
Looks like the issue is here: https://github.com/craftcms/generator/blob/main/src/generators/Plugin.php#L116

```php
        $this->minCraftVersion = $this->command->prompt('Minimum Craft CMS version:', [
            'default' => Craft::$app->getVersion(),
            'validator' => function(string $input, ?string &$error): bool {
                if (!preg_replace('/^[\d\.]+\(-\w+(\.\d+)?)?$/', $input)) {
                    $error = 'Invalid version.';
                    return false;
                }
                if (version_compare($input, '4.3.5', '<')) {
                    $error = 'Generated plugins must require Craft CMS 4.3.5 or later.';
                    return false;
                }
                return true;
            },
        ]);
```

Probably it should be `preg_match()` not `preg_replace()`

**Comment by khalwat** [comment]
@MarcelWeidum what you're trying to do will fail anyway, because apparently any plugins you create must be compatible with `craftcms/cms` `^4.3.5` anyway, as per this check:

```php
                if (version_compare($input, '4.3.5', '<')) {
                    $error = 'Generated plugins must require Craft CMS 4.3.5 or later.';
                    return false;
                }
```

**Comment by brandonkelly** [comment]
Thanks for reporting! Just released v1.0.2 with a fix for that error.

As @khalwat pointed out, you’ll need to keep it set to 4.3.5 at a minimum, but at least now you’ll get a normal input validation error about it.
